src/Dependencies/Collections/ImmutableSegmentedDictionary`2+PrivateMarshal.cs

```diff
@@ -40,6 +41,18 @@ internal static class PrivateInterlocked
 
                 return new ImmutableSegmentedDictionary<TKey, TValue>(dictionary);
             }
+
+            /// <inheritdoc cref="SegmentedCollectionsMarshal.GetValueRefOrNullRef{TKey, TValue}(ImmutableSegmentedDictionary{TKey, TValue}, TKey)"/>
+            public static ref readonly TValue FindValue(ImmutableSegmentedDictionary<TKey, TValue> dictionary, TKey key)
+                => ref SegmentedCollectionsMarshal.GetValueRefOrNullRef(dictionary._dictionary, key);
```

Comment:
>=> ref SegmentedCollectionsMarshal.GetValueRefOrNullRef(dictionary._dictionary, key); [](http://example.com/codeflow?start=16&length=85)

I assume you like this better than having SegmentedCollectionMarshal use AsSegmentedDictionary directly?

Reply:
Didn't think of that. I'll test it and see.

Reply:
➡️ Will leave this one. I'd remove it if this was the only method in the `PrivateMarshal` class, but there are many others.